src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -685,16 +694,23 @@ bool LibraryCallKit::inline_vector_frombits_coerced() {
   VectorMaskUseType checkFlags = (VectorMaskUseType)(is_mask ? VecMaskUseAll : VecMaskNotUsed);
   int opc = bcast_mode == VectorSupport::MODE_BITS_COERCED_LONG_TO_MASK ? Op_VectorLongToMask : Op_Replicate;
 
-  if (!arch_supports_vector(opc, num_elem, elem_bt, checkFlags, true /*has_scalar_args*/)) {
+  Node* broadcast = nullptr;
+  Node* bits = argument(3);
+  BasicType converted_elem_bt = elem_bt == T_DOUBLE ? T_LONG : (elem_bt == T_FLOAT ? T_INT: elem_bt);
+  // If bits sets or unsets all lanes, convert VectorLongToMask to Replicate.
+  if (opc == Op_VectorLongToMask &&
+      is_maskall_type(bits_type, num_elem) &&
+      arch_supports_vector(Op_Replicate, num_elem, converted_elem_bt, checkFlags, true /*has_scalar_args*/)) {
+        opc = Op_Replicate;
+        elem_bt = converted_elem_bt;
+        bits = gvn().longcon(bits_type->get_con() == 0L ? 0L : -1L);
```

Comment:
The `maskAll(false)` will be generated only if the input bits is `0L`. So if `bits = 0xffff0000`, and the `vlen = 4`, what is the expected mask? Is it expected to all false or all true? From the API design, I guess it is all false. But from the code above, it will be all true, right? Could you please double check and add a test for that?